src/sagemaker_training/smdataparallel.py

```diff
@@ -432,3 +451,32 @@ def _parse_custom_mpi_options(custom_mpi_options):
     parser.add_argument("--NCCL_DEBUG", default="INFO", type=str)
 
     return parser.parse_known_args(custom_mpi_options.split())
+
+
+def _validate_smddp_coll_libs_present():
+    """Check if SMDDP collective communication libraries are present in the
+    container and have executable permissions. These libraries are installed in
+    SageMaker DLCs via an smddp-installer-<versions>.run file in the
+    deep-learning-containers package.
+
+    libsmddp.so is preloaded for training when SMDDP Collectives are used.
+    The other libraries libsmddpcoll.so, liboutofband.so and libgloo.so are
+    essential for successful training. Thus, if the libraries are absent,
+    we want to avoid training failure by disabling SMDDP Collectives.
+    """
+    smddp_lib_paths = [
+        "/opt/conda/lib/libsmddp.so",
+        "/opt/conda/lib/libsmddpcoll.so",
```

Comment:
Could these change in the future? 
Let's make this `smddp_lib_paths`-> `SMDDP_LIB_PATHS` and move to the start of the file.

Reply:
These should not change in the near future. 
Will make it a constant.